src/core/StringX.h

```diff
@@ -28,6 +28,8 @@
 #include "../os/typedefs.h"
 #include "./PwsPlatform.h"
 
+extern void trashMemory(void *buffer, size_t length);
```

Comment:
The reason I had this as an extern here is because including "Util.h" created a circular dependancy with StringX.h. i.e. Util.h needs StringX.h and StringX.h needs Util.h. Probably a cleaner way is to separate out trashMemory into it's own .cpp/.h file. What do you think?